packages/datatrak-web/src/features/Survey/utils/utils.ts

```diff
@@ -67,3 +68,40 @@ export const getErrorsByScreen = (
     }, {}) ?? {}
   );
 };
+
+const hasEntityQuestionConfig = (
+  ssc: SurveyScreenComponent,
+): ssc is SurveyScreenComponent & {
+  config: { entity: EntityQuestionConfig };
+} =>
+  (ssc.type === QuestionType.Entity || ssc.type === QuestionType.PrimaryEntity) &&
+  ssc.config?.entity !== undefined;
+
+export const getParentQuestionId = (question: SurveyScreenComponent) =>
+  hasEntityQuestionConfig(question) && question.config.entity?.filter?.parentId?.questionId;
+
+export const getPrimaryQuestionAncestorAnswers = (question, questionsById, ancestorsByType) => {
+  const filterType = question?.config?.entity?.filter?.type[0];
+  const ancestor = ancestorsByType[filterType];
+  const record = {
+    [question.id]: ancestor.id,
+  };
+  const parentQuestionId = getParentQuestionId(question);
+  if (!parentQuestionId) {
+    return record;
+  }
+  const parentQuestion = questionsById[parentQuestionId];
+  return {
+    ...record,
+    ...getPrimaryQuestionAncestorAnswers(parentQuestion, questionsById, ancestorsByType),
+  };
+};
+
+export const getPrimaryEntityParentQuestionIds = (primaryEntityQuestion, questions) => {
```

Comment:
```suggestion
export const getPrimaryEntityParentQuestionIds = (entityQuestion, questions) => {
```

This is nitpicky sorry, but makes more sense I think